### PR TITLE
Replace any and unknown types with existing ones in the static/pane directory

### PR DESCRIPTION
**Pull request by ArtAhmetaj**
This PR adds the existing types: CompilerInfo, CompilationResult to the method signatures of pane and its implementations,and converts the options parameter to string instead of unknown conforming the event-map types.
Properties were added in both types to make existing code compliant.
This PR references issue: #4365

### CODE REVIEW

File: types/compilation/compilation.interfaces.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ export type CompilationResult = {
     downloads?: BuildEnvDownloadInfo[];
     gccDumpOutput?: any;
     languageId?: string;
-
+    source?: string;
```
Comment (partouf) [inline types/compilation/compilation.interfaces.ts:62]: this looks like a bug, in what situation is this used?

Comment (partouf) [reply types/compilation/compilation.interfaces.ts:62]: Found the issue, will make a separate bugreport for it. You don't have to fix it.

**Comment by RubenRBS** [comment]
Thanks for taking the time to implement the needed changes :)

Locally, you can run `make check` (Or `npm run check` if you don't have `make` available) to run the linter, the formatter and the tests, so that you can check what's failing locally. Do let us know if you have any problem fixing those issues and we'll gladly help

**Comment by RubenRBS** [comment]
Some more issues remain, but it should be easy to fix them. Let us know otherwise!

**Pull request closed without merging**